client/components/web-preview/index.jsx

```diff
@@ -135,11 +142,16 @@ const WebPreview = React.createClass( {
 
 	keyDown( event ) {
 		if ( event.keyCode === 27 ) {
-			this.props.onClose();
+			this.onClose();
 			event.preventDefault();
 		}
 	},
 
+	close() {
+		this.props.setPreviewShowing( false );
```

Comment:
I wonder if we can put this in the `componentWillUmount()` function, instead of hijacking `this.props.onClose`